app/src/main/java/com/poupa/vinylmusicplayer/provider/MusicPlaybackQueueStore.java

```diff
@@ -208,15 +210,26 @@ private synchronized void saveQueue(final String tableName, @NonNull final Array
 
 
     @NonNull
-    private ArrayList<IndexedSong> getSongPosition(@Nullable Cursor cursor, @NonNull final ArrayList<Song> songs) {
+    private ArrayList<IndexedSong> getSongPosition(@Nullable Cursor cursor, @NonNull final ArrayList<Song> songs, @NonNull final ArrayList<Long> removedSongIds) {
         ArrayList<IndexedSong> queue = new ArrayList<>();
 
         if (cursor != null && cursor.moveToFirst()) {
             int i = 0;
-            int idColumns = cursor.getColumnIndex(MusicPlaybackColumns.INDEX_IN_QUEUE);
+            int idColumn = cursor.getColumnIndex(BaseColumns._ID);
+            int indexColumn = cursor.getColumnIndex(MusicPlaybackColumns.INDEX_IN_QUEUE);
+
             do {
-                queue.add(new IndexedSong(songs.get(i), cursor.getInt(idColumns), IndexedSong.INVALID_INDEX));
-                i++;
+                long songId = cursor.getLong(idColumn);
+                int songIndex = cursor.getInt(indexColumn);
+
+                if (removedSongIds.contains(songId)) {
+                    // Add a place holder song here, to be removed after by the caller
+                    // This is done to maintain consistent queue and playing position
+                    queue.add(new IndexedSong(Song.EMPTY_SONG, songIndex, IndexedSong.INVALID_INDEX));
+                } else {
+                    queue.add(new IndexedSong(songs.get(i), songIndex, IndexedSong.INVALID_INDEX));
```

Comment:
Continuation of previous comment:
 it seems to me like the real crash come from this line "get(i)", it crash because there is a split between songs+removedSongIds size and the cursor.next/i++ done here when the removeSongIds is not attached. 
Morever, conceptually should this function really do an operation on this cursor to redo what the StoreLoader and discography already did ? 
It seems that for stability/cleaner code getting the index_in_queue should be done at the same time the orphanIdsCleaner is defined or at least that we don't check that removedSongIds contains something (as it could not have been attached) but that songs contains the songId

Reply:
Feel free to propose a PR to improve this 😀

From my point of view, we need to restructure the queue DB, or even incorporate it and the history DB with the Discography so that the concepts of orphan/zombie/stale dont leak to other classes.

I might have a look at that idea, but for now let's tackle the crashes first.

Reply:
Ok, could you just at least temporarily add in the if condition a check on the size of songs:
If (removedSongIds.contains(songId) || i >= songs.size()) {

Reply:
ok